File: Programmers/Lv0/주사위게임2.cpp
#include <string>
#include <vector>
#include <cmath>
#include <set>
using namespace std;

int solution(int a, int b, int c) {
    if (a == b && b == c)
        return (3 * a) * (3 * pow(a, 2)) * (3 * pow(a, 3));
    else if (a != b && b != c && a != c)
        return a + b + c;
    else
        return (a + b + c) * (pow(a, 2) + pow(b, 2) + pow(c, 2));
}

//  중복을 허용하지 않는 set을 이용한 방법
int solution(int a, int b, int c)
{
    set<int> s{a, b, c};
    if(s.size() == 3)
        return a + b + c;
    if(s.size() == 2)
        return (a + b + c) * (pow(a, 2) + pow(b, 2) + pow(c, 2));
    if(s.size() == 1)
        return (a + b + c) * (pow(a, 2) + pow(b, 2) + pow(c, 2)) * (pow(a, 3) + pow(b, 3) + pow(c, 3));

}

File: Programmers/Lv0/주사위게임2.py
def solution(a, b, c):
    if (a == b and b == c):
        return (3 * a) * (3 * pow(a, 2)) * (3 * pow(a, 3))
    elif (a != b and b != c and a != c):
        return a + b + c
    else:
        return (a + b + c) * (pow(a, 2) + pow(b, 2) + pow(c, 2))
    
# 중복을 허용하지 않는 set의 특징을 이용한 방법
def solution(a, b, c):
    check = len(set([a, b, c]))
    if check == 1:
        return 3*a*3*(a**2)*3*(a**3)
    elif check == 2:
        return (a+b+c)*(a**2+b**2+c**2)
    else:
        return (a+b+c)